dashboard/test/controllers/pd/workshop_material_orders_controller_test.rb

```diff
@@ -0,0 +1,107 @@
+require 'test_helper'
+class Pd::WorkshopMaterialOrdersControllerTest < ::ActionController::TestCase
+  self.use_transactional_test_case = true
+  setup_all do
+    @workshop = create :pd_ended_workshop, course: Pd::Workshop::COURSE_CSF
+    @teacher = create :teacher
+    @enrollment = create :pd_enrollment, :from_user, user: @teacher, workshop: @workshop
+  end
+
+  setup do
+    Pd::Enrollment.any_instance.stubs(completed_survey?: true)
+  end
+
+  test_redirect_to_sign_in_for :new, params: -> {{enrollment_code: @enrollment.code}}
+
+  test 'teacher without survey is redirected to survey' do
```

Comment:
Same. The survey page itself will render a 404 in that case [here](https://github.com/code-dot-org/code-dot-org/blob/0eff94d25852efc15eb5d6721df4916e224c9c3c/pegasus/sites.v3/code.org/public/pd-workshop-survey/splat.haml#L12). I'm not too concerned about this path being a great experience because someone would have to construct it from an enrollment code. I just want to be sure they can't fill out any of the surveys if they aren't supposed to, which is covered.